include/reactor-uc/macros.h

```diff
@@ -28,12 +28,27 @@
  * @brief Schedule an event on an action
  *
  */
-#define lf_schedule(action, val, offset)                                                                               \
+
+#define lf_schedule_with_val(action, offset, val)                                                                      \
   do {                                                                                                                 \
     __typeof__(val) __val = (val);                                                                                     \
-    (action)->super.schedule(&(action)->super, (offset), (const void *)&__val);                                        \
+    lf_ret_t ret = (action)->super.schedule(&(action)->super, (offset), (const void *)&__val);                         \
+    if (ret == LF_FATAL) {                                                                                             \
+      throw("no memory in action!");                                                                                   \
```

Comment:
I would say that it is not a fatal error that schedule fails. It might happen if this is a physical action and the runtime is lagging too much behind. But can we return a value from this `lf_schedule_with_val` macro? There might be some GCC extension allowing it.

Reply:
`LF_FATAL` is only returned if you schedule an action, that has no value with a value.

Reply:
OK, then I will update the error message to be a little more descriptive >P